File: entities/src/main/java/org/fortunerise/entities/Game.java
package org.fortunerise.entities;


import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "games")
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "date")
    @Temporal(TemporalType.DATE)
    private Date date;

    @Column(name = "roll")
    private Integer roll;

    @Column
    private BigDecimal payout;

    @OneToMany(mappedBy = "game", cascade = CascadeType.ALL)
    private List<Bet> bets;

    public Game() {}

    public Game(Date date, Integer roll, User user) {
        this.date = date;
        this.roll = roll;
        this.user = user;
    }

    public Integer getId() { return id; }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Integer getRoll() {
        return roll;
    }

    public void setRoll(Integer roll) {
        this.roll = roll;
    }

    public BigDecimal getPayout() {
        return payout;
    }

    public void setPayout(BigDecimal payout) {
        this.payout = payout;
    }
}

File: fortune-rise-frontend/src/app/games/models/game.ts
import {Bet} from './bet';

export class Game {
  gameId!: number;
  payout?: number;
  roll?: number;
  date?: Date;
  userId?: number;
  bets?: Bet[]

  constructor(data?: Partial<Game>) {
    if (data) {
      this.gameId = data.gameId!;
      this.roll = data.roll;
      this.payout = data.payout;
      this.date = new Date(data.date!);
      this.userId = data.userId;
      this.bets = data.bets;
    }
  }
}
